dynamicbanscore.go

```diff
@@ -0,0 +1,142 @@
+// Copyright (c) 2016 The btcsuite developers
+// Use of this source code is governed by an ISC
+// license that can be found in the LICENSE file.
+
+package main
+
+import (
+	"fmt"
+	"math"
+	"sync"
+	"time"
+)
+
+const (
+	// Halflife defines the time (in seconds) by which the transient part
+	// of the ban score decays to one half of it's original value.
+	Halflife = 60
+	lambda   = math.Ln2 / Halflife
```

Comment:
Alternatively, I could simply inline it, it is used at two locations. Let me know which one is preferred and I will make the change.